lib/vm/src/instance.rs

```diff
@@ -1163,9 +1163,9 @@ fn get_memory_init_start(init: &DataInitializer<'_>, instance: &Instance) -> usi
     if let Some(base) = init.location.base {
         let val = unsafe {
             if let Some(def_index) = instance.module.local_global_index(base) {
-                *instance.global(def_index).as_u32()
+                instance.global(def_index).to_u32()
```

Comment:
I'm surprised we use a `u32` here. I understand that we need a `u32`, but Wasm doesn't support `u32`, only `i32`. Did we “cheat” by storing random/non-standard bytes in a Global?

Reply:
Wasm's `I32` is neither rust's `i32` nor rust's `u32`. It is 32 bits long, but it doesn't include whether the number is signed or unsigned in a division, for example, while rust's i32/u32 does.

Reply:
OK.